src/types.ts

```diff
@@ -877,6 +877,26 @@ export interface BrowseMediaSource {
   children_media_class?: string | null;
   thumbnail: string | null;
   children?: BrowseMediaSource[] | null;
+  frigate?: FrigateBrowseMediaMetadata;
+}
+
+export interface FrigateBrowseMediaMetadata {
+  event: FrigateEvent;
+}
```

Comment:
Funny! I have a PR brewing that also adds these, although in a slightly different way:

```ts
interface BrowseMediaSource {
  title: string;
  media_class: string;
  media_content_type: string;
  media_content_id: string;
  can_play: boolean;
  can_expand: boolean;
  children_media_class?: string | null;
  thumbnail: string | null;
  children?: BrowseMediaSource[] | null;
}

export interface FrigateBrowseMediaSource extends BrowseMediaSource {
  children?: FrigateBrowseMediaSource[] | null;
  frigate?: {
    event: {
      camera: string;
      end_time: number;
      false_positive: boolean;
      has_clip: boolean;
      has_snapshot: boolean;
      id: string;
      label: string;
      start_time: number;
      top_score: number;
      zones: string[];
    };
  };
}

export const frigateBrowseMediaSourceSchema: z.ZodSchema<BrowseMediaSource> = z.lazy(
  () =>
    z.object({
      title: z.string(),
      media_class: z.string(),
      media_content_type: z.string(),
      media_content_id: z.string(),
      can_play: z.boolean(),
      can_expand: z.boolean(),
      children_media_class: z.string().nullable().optional(),
      thumbnail: z.string().nullable(),
      children: z.array(frigateBrowseMediaSourceSchema).nullable().optional(),
      frigate: z
        .object({
          event: z.object({
            camera: z.string(),
            end_time: z.number().nullable(),
            false_positive: z.boolean(),
            has_clip: z.boolean(),
            has_snapshot: z.boolean(),
            id: z.string(),
            label: z.string(),
            start_time: z.number(),
            top_score: z.number(),
            zones: z.string().array(),
          }),
        })
        .optional(),
    }),
);
```

I think this is slightly nicer as it leaves `BrowseMediaSource` alone and extends it, it also overrides `children` to show that children of frigate media are always other frigate media. Breaking out FrigateEvent (as you have) may be better though.

Reply:
I agree with all you said, I'll definitely pull in your changes once your PR is merged to `4.0.0` and use those